### PR TITLE
Adding support for file title without renaming file, updating docker files.

### PR DESCRIPTION
**Pull request by JunioCalu**

### CODE REVIEW

**Comment by jb-alvarado** [comment]
Thanks @JunioCalu! I would like to have title optional, Playlist should serialize and de-serialize with and without the title key.

**Comment by JunioCalu** [comment]
In my tests, the old .json playlists that did not contain the 'title' field were running normally. In other words, the new 'function' exhibited a pseudo-optional behavior.

File: ffplayout-api/src/api/routes.rs [hunk 1]
```diff
@@ -132,6 +132,7 @@ fn time_before() -> NaiveDateTime {
 struct ProgramItem {
     source: String,
     start: String,
+    title: String,
```
Comment (jb-alvarado) [inline ffplayout-api/src/api/routes.rs]: I would like to have here a `Option<String>`.

File: ffplayout-engine/src/utils/mod.rs [hunk 2]
```diff
@@ -239,6 +239,7 @@ pub fn prepare_output_cmd(
 /// map media struct to json object
 pub fn get_media_map(media: Media) -> Value {
     json!({
+        "title": media.title,
```
Comment (jb-alvarado) [inline ffplayout-engine/src/utils/mod.rs:255]: Maybe here is also a way to make title optional.

File: lib/src/utils/json_serializer.rs [hunk 3]
```diff
@@ -73,6 +74,7 @@ pub fn set_defaults(playlist: &mut JsonPlaylist) {
     for (i, item) in playlist.program.iter_mut().enumerate() {
         item.begin = Some(start_sec);
         item.index = Some(i);
+        item.title = item.title.clone();
```
Comment (jb-alvarado) [inline lib/src/utils/json_serializer.rs]: this is useless :-).

File: lib/src/utils/mod.rs [hunk 4]
```diff
@@ -65,6 +65,7 @@ pub struct Media {
 
     #[serde(skip_serializing, skip_deserializing)]
     pub index: Option<usize>,
+    pub title: String,
```
Comment (jb-alvarado) [inline lib/src/utils/mod.rs]: Here also a `Option<String>` and on top a: `#[serde(skip_serializing_if = "Option::is_none")]`

File: lib/src/utils/mod.rs [hunk 5]
```diff
@@ -138,6 +139,7 @@ impl Media {
         Self {
             begin: None,
             index: Some(index),
+            title: String::new(),
```
Comment (jb-alvarado) [inline lib/src/utils/mod.rs]: Needs then be a `None`

File: ffplayout-engine/src/utils/mod.rs (hunk 2 above)
Comment (JunioCalu) [reply ffplayout-engine/src/utils/mod.rs:255]: I have finished the corrections, sorry for the delay. If there are more corrections, please mention them.

File: ffplayout-api/src/api/routes.rs (hunk 1 above)
Comment (JunioCalu) [reply ffplayout-api/src/api/routes.rs]: I have finished the corrections, sorry for the delay. If there are more corrections, please mention them.

**Comment by jb-alvarado** [comment]
Thank you for the changes!

Please have a bit patience, I would like to add this in version 0.22. But at the moment I'm working on SSE to replace the status from system and playout.

**Comment by JunioCalu** [comment]
Great, I'll wait for version 0.22 then.

**Pull request merged**